service/worker/workerdeployment/workflow.go

```diff
@@ -908,14 +909,17 @@ func (d *WorkflowRunner) syncVersion(ctx workflow.Context, targetVersion string,
 func (d *WorkflowRunner) syncUnversionedRamp(ctx workflow.Context, versionUpdateArgs *deploymentspb.SyncVersionStateUpdateArgs) error {
 	activityCtx := workflow.WithActivityOptions(ctx, defaultActivityOptions)
 
-	// DescribeVersion activity to get all the task queues in the current version
+	// DescribeVersion activity to get all the task queues in the current version, or the ramping version if current is nil
+	version := d.State.RoutingConfig.CurrentVersion //nolint:staticcheck // SA1019: worker versioning v0.31
+	if version == worker_versioning.UnversionedVersionId {
+		version = d.State.RoutingConfig.RampingVersion //nolint:staticcheck // SA1019: worker versioning v0.31
```

Comment:
why are we suppressing this lint role, and why is it specific to versioning v0.31?